### PR TITLE
Fix backward operator return values to respect requires_grad settings

### PR DESCRIPTION
**Pull request by guozixu2001**
## Motivation and Context

There are some error when running resnet50 testcase

image

image

This change addresses an issue with the `conv2d_backward` function where it was returning unnecessary gradients even when `requires_grad` was not set for some parameters. By respecting the requires_grad settings, we ensure that only the required gradients are computed and returned. 

## Description

- Modified the `backward operator` function to ensure that gradients are only included in the return dictionary if `requires_grad` is set to True for the respective parameters.
- Updated the test template to correctly handle and pass `requires_grad` information from function_paras.
- Updated the `diopiTensor` class definition to include the `requires_grad` attribute.

## Checklist
**Before PR**:

- [x] I have read and followed the workflow indicated in the [Contributors.md](https://github.com/DeepLink-org/DIOPI/blob/main/Contributors.md) to create this PR.
- [x] Pre-commit or linting tools indicated in [Contributors.md](https://github.com/DeepLink-org/DIOPI/blob/main/Contributors.md) are used to fix the potential lint issues.
- [x] Bug fixes are covered by unit tests, the case that causes the bug should be added in the unit tests.
- [x] New functionalities are covered by complete unit tests. If not, please add more unit test to ensure the correctness.
- [x] The documentation has been modified accordingly, including docstring or example tutorials.

**After PR**:

- [x] CLA has been signed and all committers have signed the CLA in this PR.

### CODE REVIEW

**Review by caikun-pjlab — APPROVED** [review]

File: diopi_test/python/conformance/diopi_functions.py [hunk 1]
```diff
@@ -382,7 +382,7 @@ def silu_backward(input, grad_outputs, **kwargs) -> Tensor:
     func = check_function("diopiSiluBackward")
     ret = func(input.context(), grad_input, grad_outputs[0], input)
     check_returncode(ret)
-    return {"input": grad_input}
+    return {"input": grad_input} if grad_input.requires_grad else {}
```
Comment (yangbofun) [inline diopi_test/python/conformance/diopi_functions.py:385]: 这里不用修改， requires_grad决定了是否求导，不求导的话，不用去调用backward。
